### ISSUE TITLE
Ability to rewrite HTTP status

### ISSUE THREAD

**Issue by tuxpizza** [joseluisq/static-web-server#268]
**Is your feature request related to a problem? Please describe.**
Would like to be able to rewrite HTTP status so present a maintenance page giving a 503.

**Describe the solution you'd like**
Would like the ability to force the index.html to throw a 503 (or maybe any other error) just for hosting a maintenance page. Via Env vars would be preferable due to the constraints I'm under since I'm using a TrueNAS Scale helms chart (don't have as much access to the configuration)

**Describe alternatives you've considered**
Just about everything, but the way my setup is right now I have no ability to use a different web server since I need traefik's ingress and am limited to the selection of helms charts that work on truenas, and caddy (downstream reverse proxy) can't load balance while rewriting HTTP status codes.

**Comment by joseluisq** [comment]
Because maintenance is a frequent task, I think that a `--maintenance-mode` feature flag with an additional custom status flag could work.

Here is just an idea.

```env
# required envs
SERVER_MAINTENANCE_MODE=true
SERVER_MAINTENANCE_MODE_STATUS=503
# optional env
SERVER_MAINTENANCE_MODE_FILE="./my_maintenance.html"
```

Let me know if this is what you are interested in.

**Comment by tuxpizza** [comment]
That's a great idea to do a maintenance flag and customizing the status code. And I'm assuming if the optional  `SERVER_MAINTENANCE_MODE_FILE` env var isn't there it'll just display the index.html but with a 503?

**Comment by joseluisq** [comment]
> I'm assuming if the optional `SERVER_MAINTENANCE_MODE_FILE` env var isn't there it'll just display the index.html but with a 503?

No, it won't.
I thought that when using the defaults (`SERVER_MAINTENANCE_MODE=true`) we could display a generic message like `The server is under maintenance mode.` with a `503` status.
So we could even make the status optional too.

For example, I propose the following.

```env
# required env, false by default
SERVER_MAINTENANCE_MODE=true

# optional env, 503 by default
SERVER_MAINTENANCE_MODE_STATUS=503

# optional env, empty by default, server uses a generic message instead
SERVER_MAINTENANCE_MODE_FILE="./my_maintenance.html"
```

**Comment by tuxpizza** [comment]
> No, it won't. I thought that when using the defaults (`SERVER_MAINTENANCE_MODE=true`) we could display a generic message like `The server is under maintenance mode.` with a `503` status. So we could even make the status optional too.
>

That makes more sense. Looks like a perfect simple solution!

**Comment by joseluisq** [comment]
Resolved in [v2.23.0](https://github.com/static-web-server/static-web-server/releases/tag/v2.23.0)
